File: cpp/CodeRetreat.cpp
#include <algorithm>
#include <vector>
#include <string>

class CodeRetreat {
  public:
    static bool is_palindrome(std::string a) {
      std::string b = a;
      std::reverse(b.begin(), b.end());
      return a.compare(b) == 0;
    }
    static int find_min(std::vector<int>& a) {
      if (a.size() == 0) {
        return -1;
      }
      std::sort(a.begin(), a.end());
      return a[0];
    }
};

File: go/coderetreat.go
package coderetreat

import (
	"fmt"
	"sort"
)

func isPalindrome(a string) bool {
	for i := 0; i < len(a)/2; i++ {
		if a[i] != a[len(a)-i-1] {
			return false
		}
	}
	return true
}

func findMin(a []int) (int, error) {
	if len(a) == 0 {
		return 0, fmt.Errorf("slice is empty")
	}
	sort.Slice(a, func(i, j int) bool {
		return a[i] < a[j]
	})
	return a[0], nil
}
